src/typedefs.jl

```diff
@@ -0,0 +1,21 @@
+@userplot CornerPlot
```

Comment:
This is a nice and easy implementation of keeping these in RecipesBase. BUT it drastically increases the number of exported names from the package (It's normally only `apply_recipe`, `userplot`, `shorthands` and `series`), thus increasing the risk of clashes. What about just defining the type, then having `@userplot` look in RecipesBase for the type and register the rest of the functionality?